File: src/main/java/fun/bookish/vertx/activemq/client/producer/ActiveMQProducer.java
package fun.bookish.vertx.activemq.client.producer;


import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;
import io.vertx.core.json.JsonObject;

public interface ActiveMQProducer {

    String getKey();

    void send(JsonObject message);

    void send(JsonObject message,Handler<AsyncResult<Void>> handler);

    void close();

    void close(Handler<AsyncResult<Void>> handler);

}

File: src/main/kotlin/cn/booklish/vertx/activemq/client/producer/ActiveMQProducer.kt
package cn.booklish.vertx.activemq.client.producer

import io.vertx.core.AsyncResult
import io.vertx.core.Handler
import io.vertx.core.json.JsonObject


interface ActiveMQProducer {

    val key:String

    fun send(message: JsonObject)

    fun send(message: JsonObject,handler:Handler<AsyncResult<Void>>?)

    fun close()

    fun close(handler:Handler<AsyncResult<Void>>)

}